include/thread_pool/mpmc_bounded_queue.hpp

```diff
@@ -169,7 +208,33 @@ inline MPMCBoundedQueue<T>& MPMCBoundedQueue<T>::operator=(MPMCBoundedQueue&& rh
 
```

Comment:
Also in general it's a good practice to set rhs to some sensible state after moving from it. This may help catching bugs when you buy mistake try to use the object which was already moved:
A a;
A a2 = std::move(a);
do something with a

If you handle the move properly and throw a proper exception when trying to use a moved object you can catch such bugs in place of use

Reply:
Fixed in #3. Removed this altogether.